jwst/outlier_detection/utils.py

```diff
@@ -0,0 +1,208 @@
+                elem1[elem2] = np.nan
+
+        del badmasks
+
+        # For a stack of images with "bad" data replaced with Nan
+        # use np.nanmedian to compute the median.
+        with warnings.catch_warnings():
+            warnings.filterwarnings(action="ignore",
+                                    message="All-NaN slice encountered",
+                                    category=RuntimeWarning)
+            median_image[row1:row2] = np.nanmedian(resampled_sci, axis=0)
+        del resampled_sci, resampled_weight
+
+    return median_image
+
+
+def compute_weight_threshold_container(resampled_models, maskpt):
+    '''
+    Compute weight means without keeping datamodels for each input open
+
+    Parameters
+    ----------
+    resampled_models : ~jwst.datamodels.ModelContainer
+        The input data models.
+
+    maskpt : float
+        The percentage of the mean weight to use as a threshold for masking.
+
+    Returns
+    -------
+    list
+        The weight thresholds for each integration.
+    '''
+
+    # Start by ensuring that the ModelContainer does NOT open and keep each datamodel
+    ropen_orig = resampled_models._return_open
+    resampled_models._return_open = True
+    # keep track of resulting computation for each input resampled datamodel
+    weight_thresholds = []
+    # For each model, compute the bad-pixel threshold from the weight arrays
+    for resampled in resampled_models:
+        weight = resampled.wht
+        weight_threshold = compute_weight_threshold(weight, maskpt)
+        weight_thresholds.append(weight_threshold)
+        # close and delete the model, just to explicitly try to keep the memory as clean as possible
+        resampled.close()
+        del resampled
+    # Reset ModelContainer attribute to original value
+    resampled_models._return_open = ropen_orig
+    return weight_thresholds
+
+
+def flag_crs_in_models(
+    input_models,
+    median_data,
+    median_wcs,
+    snr1,
+    snr2,
+    scale1,
+    scale2,
+    backg,
+    resample_data,
+):
+    for image in input_models:
+        if resample_data:
+            if 'SPECTRAL' not in image.meta.wcs.output_frame.axes_type:
+                input_pixflux_area = image.meta.photometry.pixelarea_steradians
+                input_pixel_area = compute_image_pixel_area(image.meta.wcs)
+                pix_ratio = np.sqrt(input_pixflux_area / input_pixel_area)
+            else:
+                pix_ratio = 1.0
+
+            blot = gwcs_blot(median_data, median_wcs, image.data.shape, image.meta.wcs, pix_ratio)
+        else:
+            blot = median_data
+
+        # dq flags will be updated in-place
+        flag_resampled_model_crs(image, blot, snr1, snr2, scale1, scale2, backg, resample_data)
+
+
+def flag_resampled_model_crs(
+    image,
+    blot,
+    snr1,
+    snr2,
+    scale1,
+    scale2,
+    backg,
+    resample_data,
+):
+    # previous versions of the code generated all of the blot models and wrote
```

Comment:
I'd rather leave this as a comment. Changing it to a triple-quote would turn it into a docstring (since it's the first line in the function). I'd also be ok just removing it as it provides little outside the context of this PR.

Reply:
removing sounds good

Reply:
Thanks! I removed the comment but then that shifted the next comment up 🤦 So I added a very simple docstring:
69a5e4e190084f4f4ab8e9a44bba303a47fa66c0